chia/cmds/init_funcs.py

```diff
@@ -480,14 +477,14 @@ def chia_init(
 
             save_config(root_path, "config.yaml", config)
 
-        else:
-            config = load_config(root_path, "config.yaml", acquire_lock=False)["full_node"]
-            db_path_replaced = config["database_path"].replace("CHALLENGE", config["selected_network"])
-            db_path = path_from_root(root_path, db_path_replaced)
-            mkdir(db_path.parent)
+    else:
+        config = load_config(root_path, "config.yaml")["full_node"]
```

Comment:
Loading config not under lock here

Reply:
The `load_config()` function does acquire the lock inside.  The intent was to work towards all 'external' functions just getting the lock.  Saving is the remaining one that does not.

Reply:
OK got it